Client/Frontend/Browser/OpenWithSettingsViewController.swift

```diff
@@ -0,0 +1,134 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+class OpenWithSettingsViewController: UITableViewController {
+
+    private lazy var mailSchemes: [[String:String]] = {
+        var plistSchemes: [[String:String]] = self.loadMailSchemes()
+        return plistSchemes
+    }()
+    private var mailSchemeEnableCache: [String:Bool] = [String:Bool]()
+    private let prefs: Prefs
+    private var currentChoice: String = "mailto"
+
+    private let BasicCheckmarkCell = "BasicCheckmarkCell"
+
+    init(prefs: Prefs) {
+        self.prefs = prefs
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = Strings.SettingsOpenWithSectionName
+
+        tableView.accessibilityIdentifier = "OpenWithPage.Setting.Options"
+
+        tableView.registerClass(UITableViewCell.self, forCellReuseIdentifier: BasicCheckmarkCell)
+        tableView.backgroundColor = UIConstants.TableViewHeaderBackgroundColor
+
+        let headerFooterFrame = CGRect(origin: CGPointZero, size: CGSize(width: self.view.frame.width, height: UIConstants.TableViewHeaderFooterHeight))
+        let headerView = SettingsTableSectionHeaderFooterView(frame: headerFooterFrame)
+        headerView.titleLabel.text = Strings.SettingsOpenWithPageTitle
+        headerView.showTopBorder = false
+        headerView.showBottomBorder = true
+
+        let footerView = SettingsTableSectionHeaderFooterView(frame: headerFooterFrame)
+        footerView.showTopBorder = true
+        footerView.showBottomBorder = false
+
+        tableView.tableHeaderView = headerView
+        tableView.tableFooterView = footerView
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(OpenWithSettingsViewController.SELappDidBecomeActiveNotification), name: UIApplicationDidBecomeActiveNotification, object: nil)
+    }
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        reloadMailSchemeEnableCache()
+        if let prefMailtoScheme = self.prefs.stringForKey("MailToOption"), let previousChoice = mailSchemeEnableCache[prefMailtoScheme], let defaultScheme = mailSchemes[0]["scheme"] {
+            self.currentChoice = previousChoice ? prefMailtoScheme : defaultScheme
+        }
+        tableView.reloadData()
+    }
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        self.prefs.setString(currentChoice, forKey: "MailToOption")
+    }
+
+    func SELappDidBecomeActiveNotification() {
```

Comment:
NIt: Don't need to prefix with SEL anymore since selectors are typed. You can just use `appDidBecomeActive()`.